core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/runtime/BaseRuntime.java

```diff
@@ -52,18 +53,31 @@ protected Monitor getMonitor() {
     }
 
     /**
-     * Main entry point to runtime initialization. Calls all methods.
+     * Main entry point to runtime initialization. Calls all methods
+     * and sets up a context shutdown hook at runtime shutdown.
      */
     protected void boot() {
+        boot(true);
+    }
+
+    /**
+     * Main entry point to runtime initialization. Calls all methods.
+     *
+     * @param addShutdownHook if {@code true}, sets up a context shutdown hook at runtime shutdown
+     */
+    protected void boot(boolean addShutdownHook) {
```

Comment:
I am not sure it is a good idea to remove shutdown hooks just for the sake of reducing output in tests. I think it is worth having test executing this piece of code as well. Did you evaluate any other solution? How verbose was the output of this part?

Reply:
it's writing "Shutdown complete" to stdout after test runtime is completed, and streams are not intercepted anymore.

Why should there be a shutdown hook specifically for these resources and not for others? Shutdown hooks is not how you manage resources in tests.

Reply:
Ok. Why not moving the shutdown hook initialization to the main method to prevent the [flag argument](https://martinfowler.com/bliki/FlagArgument.html)?

Reply:
Because the shutdown hook is configured in the middle of the `boot` method